### ISSUE TITLE
SwiftUI code format question

### ISSUE THREAD

**Issue by hapton** [nicklockwood/SwiftFormat#1436]
How can we make swiftFormat formatting
  Text("ssssssssssss").background(Color(hex: 0x333333)).font(Font.caption).padding(.horizontal, 8).padding(.trailing, 8)
to
  Text("ssssssssssss")
      .background(Color(hex: 0x333333))
      .font(Font.caption)
      .padding(.horizontal, 8)
      .padding(.trailing, 8)

This is swiftUI code.Code wrapping looks clearer。

**Comment by nicklockwood** [comment]
@hapton SwiftFormat doesn't currently have a rule for this I'm afraid (mainly because I'm still not sure what the best way to format SwiftUI is myself!)

**Comment by Parabak** [comment]
@nicklockwood we have the same question. With one more use-case:
```swift
- }.padding(.horizontal, .xSmall)
+ }
   .padding(.horizontal, .xSmall)
```

Will you accept the pull request if it will be done as Opt-in Rule (disabled by default) ?

**Comment by nicklockwood** [comment]
@Parabak I'm not sure I understand the proposes rule here - you want to indent all chained methods after a } by one additional space?

**Comment by Parabak** [comment]
All SwiftUI modifiers:
* should be preceded with line break 
* apply indent if modifier **is not** preceded with `}`

**Comment by nicklockwood** [comment]
@Parabak I think the last one is already covered by the [emptyBraces](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#emptyBraces) rule (unless you want different behavior when the braces are part of a method chain?)

**Comment by Parabak** [comment]
@nicklockwood with last examples I tried to illustrate the fact that after `)` modifier `textLinkAction` starts from new line and without indent.

regarding naming: maybe `--wrapSwiftUIModifiers` ? either enabled or disabled.

**Comment by nicklockwood** [comment]
@Parabak I don't think this has anything to do with SwiftUI per-se, it's just more common to see these method chains in SwiftUI code than other places.

Probably a new rule like `wrapChainedMemberAccess` or something. I'll think about it.

**Comment by facumenzella** [comment]
I've created this [issue](https://github.com/nicklockwood/SwiftFormat/issues/490) a while ago. In SwiftLint is called `multiline-function-chains`.

I was planning to work on this at some point this month =)

**Comment by nicklockwood** [comment]
@Parabak it's probably feasible to add a swiftui-only option once the rule is done, but you could also manage that by using a separate .swiftformat file for your Views subdirectory.

**Comment by Parabak** [comment]
Right, the thing is that we have around 30 such subdirectories. Most likely we can use a symlink there, but having this as an option will simplify setup. 

@facumenzella, if you need help, please ping me. I can add `swiftui-only` option by my own, as @nicklockwood suggested.
